crates/rust-analyzer/src/flycheck.rs

```diff
@@ -69,6 +119,28 @@ impl CargoOptions {
     }
 }
 
+/// The flycheck config from a rust-project.json file
+#[derive(Debug, Default)]
+pub(crate) struct FlycheckConfigJson {
+    // XXX: unimplemented because not all that important> nobody
+    // doing custom rust-project.json needs it most likely
+    // pub workspace_template: Option<project_json::Runnable>,
```

Comment:
Handle all the flychecks where we would normally run `cargo check --workspace`. Currently we pass these requests to the flycheck handle, but always end up doing nothing, because the saved file is None and there is no label to interpolate.

rust-project doesn't have anything to run for this, but it is not that hard to add. I have some simple BXL that builds and prints out the diag.txt/json for a target and all its dependencies, and frequently run this on `//...`. The only problem with it is that the BXL is slow at unifying all the deps of all the targets and always stutters for a few seconds doing "graph computations", but I'm probably doing something dumb. Other than that works exactly like cargo check.